doc/releases/changelog-dev.md

```diff
@@ -3,6 +3,20 @@
 # Release 0.20.0-dev (development release)
 
 <h3>New features since last release</h3>
+
+* The `metric_tensor` transform can now be used to compute the full
+  tensor, beyond the block diagonal approximation. 
+  [(#1725)](https://github.com/PennyLaneAI/pennylane/pull/1725)
+
+  This is done using Hadamard tests and requires an additional wire 
+  on the device to execute the tapes produced by the transform, 
+  as compared to the number of wires required by the original tape.
+  The transform defaults to computing the full tensor, which can
+  be controlled by the `approx` keyword argument.
+  See the 
+  [qml.metric_tensor docstring](https://pennylane.readthedocs.io/en/latest/code/api/pennylane.transforms.metric_tensor.html).
+  for more information and usage details.
+
```

Comment:
I see where you are coming from but because the cost and output differ so significantly, I would not go for an implicit logic here. Users might expect to get the full metric and silently be provided only with the block diagonal because they "forgot" to add a wire to their device...

Reply:
Makes sense 🙂 Just to clarify, with the changes in this PR, then
```python
qml.metric_tensor(qnode)(*args)
```
will raise an error if the device doesn't have an additional wire?

Reply:
Yes, exactly :) :+1:

Reply:
I see; this is something I would want to avoid, since it is a significant breaking change 🤔